src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs

```diff
@@ -411,16 +411,23 @@ private void Statement(Block block)
             if (Context.DesignTimeMode || !isMarkup)
             {
                 // CODE owns whitespace, MARKUP owns it ONLY in DesignTimeMode.
-                if (lastWs != null)
+                if (lastWhitespace != null)
                 {
-                    Accept(lastWs);
+                    Accept(lastWhitespace);
                 }
             }
             else
             {
+                var nextSymbol = Lookahead(1);
+
                 // MARKUP owns whitespace EXCEPT in DesignTimeMode.
                 PutCurrentBack();
-                PutBack(lastWs);
+
+                // Don't putback the whitespace if it precedes a '<text>' tag.
+                if (!nextSymbol.Content.Equals(SyntaxConstants.TextTagName))
+                {
+                    PutBack(lastWhitespace);
+                }
```

Comment:
Two concerns:

1. This method has some wonkyness to it, for instance the two comments on line 423 and 413 completely contradict each other. Moreover, I don't see why Markup should ever own whitespace even in design time mode. That would end up breaking intellisense for C#. We may want to clarify this with @anurse for what his intentions where when writing this bit.
1. This comment could be completely null and void based on the answer to 1, but this seems like the wrong place to put logic handling the `<text>` tag. There's a few locations where we special case `<text>` in `HtmlMarkupParser.Block` but neither should be impacted by what happens in the C# statement block from a conceptual level. Therefore, IF we can allow C# to own whitespace always in statements that would remove the need for this check entirely.